In [1]:
import pandas as pd
from os.path import join as path_join

In [2]:
GI50 = pd.read_csv(path_join("data", "GI50.csv"))
chem_names = pd.read_csv(
    path_join("data", "chemnames_Aug2013.txt"),
    "|",
    header=None,
    names=["NSC", "Name", "Name Type"],
)
print(f"GI50 has shape {GI50.shape}")
print(f"Chem names has shape {chem_names.shape}")

GI50 has shape (4547757, 14)
Chem names has shape (251887, 3)


In [3]:
all_gi50_nsc = GI50["NSC"].unique()
all_chem_names_nsc = chem_names["NSC"].unique()

In [4]:
print(f"GI50 has {all_gi50_nsc.size} unique NSC number")
print(f"All chem names {all_chem_names_nsc.size} unique NSC number")

GI50 has 56220 unique NSC number
All chem names 67534 unique NSC number


Comparing the NSC number between all chem names and GI50 gives us a list of 13,159 NSC number that appear in both of them. This of the 56,220 NSC number in GI50 means we 76.6% of all experiments can not be used because of missing mapping.

In [5]:
only_gi50 = list(set(all_gi50_nsc) - set(all_chem_names_nsc))
print(f"GI50 has {len(only_gi50)} number not in chem names")
gi50_chem_names = list(set(all_gi50_nsc) - set(only_gi50))
print(f"A total of {len(gi50_chem_names)} NSC numbers are in both")

GI50 has 43061 number not in chem names
A total of 13159 NSC numbers are in both


In [6]:
filtered_names = chem_names.loc[chem_names["NSC"].isin(gi50_chem_names)]
print(f"Filtered chem names has shape {filtered_names.shape}")
filtered_names.to_csv(path_join("data", "filtered_chemnames_Aug2013.csv"))
filtered_names.head()

Filtered chem names has shape (39120, 3)


NSC             Name      Name Type
0    1     Tolylquinone  Chemical Name
1    1    p-Toluquinone  Chemical Name
2    1    Methylquinone  Chemical Name
3    1  1,4-Toluquinone  Chemical Name
4    1  2-Methylquinone  Chemical Name